vision/src/main/java/coppercore/vision/Camera.java

```diff
@@ -39,7 +38,23 @@ public CameraMeasurement getLatestMeasurement() {
     }
 
     public Matrix<N3, N1> getLatestVariance() {
+        Matrix<N3, N1> stdDev = CoreVisionConstants.singleTagStdDev;
         // TODO: Actually calculate variances!
-        return VecBuilder.fill(0.0, 0.0, 0.0);
+        double avgDistanceFromTarget = inputs.averageTagDistanceM;
+        int numTags = inputs.nTags;
+
+        if (numTags == 0) {
+            return stdDev;
+        } else {
+            if (numTags > 1) {
+                stdDev = CoreVisionConstants.multiTagStdDev;
+            } else if (numTags == 1 && avgDistanceFromTarget > 4) {
+                return CoreVisionConstants.rejectionStdDev;
+            }
+
+            // distance based variance
+            stdDev = stdDev.times(1 + (Math.pow(avgDistanceFromTarget, 2) / 30));
```

Comment:
Can we un-nest the if/else. For example
```
else if (numTags > 1) { ... }
else if (numTags == 1 && avgDistanceFromTarget > 4)
```

Also, can we make some of these magic numbers constants in CoreVisionConstants.java? In particular, 4 should be some vision cutoff constant, and the parameter(s) used in the distance based variance calculation should be made into constants in order to name them something other than "the 30 on line 56"

A comment on why 1 + x^2/30 would also be appreciated lol

Reply:
this is done, unsure why its x^2/30. i think that term can be messed around with depending on how accurate we find our vision to be from a far distance. (ie make denominator smaller to rely on it less)